### PR TITLE
My entry about the CodingKey protocol

### LINKED ISSUE #2961: [Term Entry] Swift Protocols CodingKey
### Reason/inspiration (optional)

We would like a new term entry for Swift [Protocols](https://www.codecademy.com/resources/docs/swift/protocols) about the CodingKey protocol. The entry should go in a new file in docs/content/swift/concepts/protocols/terms/codingkey/codingkey.md

The entry should
    - Have an introductory paragraph about what the protocol is and does.
    - A syntax section with a pseudo code block with an explanation of it.
    - A basic example showing the protocol in use

Please refer to the [term entry templates](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by LVC1D**
### Description (Closes #2961)

I have submitted a **Term Entry** about the `CodingKey` protocol that works in conjunction with the `Decodable` and `Encodable` protocols.

Those protocols are for the Swift objects to conform to when we want to map the key of the JSON's objects and the Swift objects' properties.

The workflow for this entry is as follows:

+ I have provided two basic pseudo-codebases for the Decodable and the Encodable scenarios
+ Then, I have proceeded with showcasing the application of the CodingKey protocol in the enumerator defined inside both the examples' structs
+ Below each example, I have shown what would the expected result be logged in the terminal upon running the examples codes

My small concern here is with the understanding of the last checkmark point about "linking any issues that are relevant to this PR". I have written to which issue number I am referring as mentioned in the parentheses.

(*Note: This is my very first PR*)

### Type of Change

- Adding a new entry **codingkey/codingkey.md** into the following path: *content/swift/concepts/protocols*

### Checklist

- [V] All writings are my own.
- [V] My entry follows the Codecademy Docs style guide.
- [V] My changes generate no new warnings.
- [V] I have performed a self-review of my own writing and code.
- [V] I have checked my entry and corrected any misspellings.
- [V] I have made corresponding changes to the documentation if needed.
- [V] I have confirmed my changes are not being pushed from my forked `main` branch.
- [V] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [V] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:1]:
```suggestion
```

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:4]:
```suggestion
Description: 'Enables the mapping of keys from JSON to a given Swift struct or class property.'
```

Comment (caupolicandiaz) [reply content/swift/concepts/protocols/terms/codingkey/codingkey.md:4]: descriptions cannot contain punctuation like apostrophes

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:22]: There should be an empty line after the note

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:23]: There should be am empty line after the syntax header

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:24]: I don't believe there needs to be a separate example but `Decodable` and `Encodable` are each just half of the `Codable` protocol which isn't mentioned

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:48]:
```suggestion
The code above demonstrates the usage of an enumerator `CodingKeys`, its cases of which refer to the JSON's object keys. It also specifies:
```

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:49]:
```suggestion
1. The raw value type of the keys (it's a `String` because all JSON keys are of a `String` type only).
```

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:50]:
```suggestion
2. The `CodingKey` protocol to which the enumerator has to conform in order to map the JSON keys with the Swift object's properties.
```

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:54]:
```suggestion
The example below will decode a JSON object array of famous musicians into a Swift model using the enumerator that conforms to the `CodingKey` protocol, and list the name and active years for each musician using a `.forEach` function.
```

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:55]:
```suggestion
```

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:57]: code indentations should use 2 spaces not tabs

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:96]:
```suggestion
> **Note:** Notice the use of the newly-created `yearsActive` property in the Swift model when JSON's key is written as `years_active`.
```

Comment (caupolicandiaz) [reply content/swift/concepts/protocols/terms/codingkey/codingkey.md:96]: docs content avoids the use of first and second person personal pronouns (I , you, we)

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:100]:
```suggestion
> **Note:** A trailing closure in the invoked function is used only to simplify the code.
```

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:110]:
```suggestion
Following the workflow of [encoding](https://www.codecademy.com/resources/docs/swift/protocols/encodable) the Swift model into a JSON data structure, the following example will display the `Exercises` struct parsed into JSON.
```

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:111]:
```suggestion
```

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:114]: update the indentations and comments should begin with a capital
// Create the CodingKyes enum to define the JSON keys

File: content/swift/concepts/protocols/terms/codingkey/codingkey.md
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/codingkey/codingkey.md:157]: add an empty line before the shell block

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@LVC1D I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by LVC1D** [comment]
@caupolicandiaz Hi again! I've refactored the entry by showcasing one cohesive example that conforms to the `Coddle` protocol. In addition, I took into account all other suggestions of yours, that included fixing the indentations, phrasings etc.

Let me know if this version looks better now. Thank you.

**Review by caupolicandiaz — APPROVED** [review]
@LVC1D This is ready for a second review

**Comment by LVC1D** [comment]
@caupolicandiaz Hi! Are there any updates on this PR, please?

**Comment by caupolicandiaz** [comment]
> @caupolicandiaz Hi! Are there any updates on this PR, please?

Please be patient. There are many PRs and few maintainers.

**Review by SSwiniarski — APPROVED** [review]
@LVC1D Dis some minor edits. Looks good now.

**Pull request merged**
